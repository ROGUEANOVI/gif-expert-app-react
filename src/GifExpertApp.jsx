import React, { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {

    const [cateogries, setCateogries] = useState([]);

    const onAddCategory = (newCategory) => {
        if (cateogries.includes(newCategory)) return;

        setCateogries([newCategory, ...cateogries]);
    }

    return (
        <>
            <h1>Gifs Expert</h1>

            <AddCategory onAddCategory={ onAddCategory } />

            {
                cateogries.map(category => (
                    <GifGrid key={ category } category={ category } />
                ))
            }
        </>
    );
};
