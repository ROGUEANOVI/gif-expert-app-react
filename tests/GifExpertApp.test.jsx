import { fireEvent, render, screen } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('Test <GifExpertApp />', () => { 
    
    const setup = () => {
        render(<GifExpertApp />);

        const input = screen.getByRole("textbox");
        const form = screen.getByRole("form");
        
        return {
          input,
          form,
        };
    };

    test('should match the snapshot', () => {
 
        const { container } = render(<GifExpertApp />);
   
        expect(container).toMatchSnapshot();
     });

    test('should show the title h1', () => { 
        
        render(<GifExpertApp />);

        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toBe('Gifs Expert');
    });

    test("Should add category if it's not repeated", () => {
        const { input, form } = setup();
        
        fireEvent.change(input, { target: { value: "Naruto" } });
        fireEvent.submit(form);

        fireEvent.change(input, { target: { value: "Saitama" } });
        fireEvent.submit(form);
     
        expect(screen.getAllByRole("heading", { level: 3 })).toHaveLength(2);
    });
     
    test("Should NOT add category if it's repeated", () => {
        const { input, form } = setup();

        fireEvent.change(input, { target: { value: "Demon Slayer" } });
        fireEvent.submit(form);

        fireEvent.change(input, { target: { value: "Demon Slayer" } });
        fireEvent.submit(form);

        fireEvent.change(input, { target: { value: "Demon Slayer" } });
        fireEvent.submit(form);
        
        expect(screen.getAllByRole("heading", { level: 3 })).toHaveLength(1);
    });
});