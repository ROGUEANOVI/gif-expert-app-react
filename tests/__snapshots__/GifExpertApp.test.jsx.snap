// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifExpertApp /> should match the snapshot 1`] = `
<div>
  <h1>
    Gifs Expert
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
